src/Orleans/Core/GrainFactory.cs

```diff
@@ -290,6 +405,35 @@ internal static TGrainInterface GetSystemTarget<TGrainInterface>(GrainId grainId
         #endregion
 
         #region Utility functions
+
+        /// <summary>
+        /// Ensures that generated code reflects the current runtime environment.
+        /// </summary>
+        private static void EnsureCodeGenerationIsCurrent()
```

Comment:
This is only need on the client side, right? Because on the silo side GrainTypeManager triggers codegen explicitly after all grain assemblies are loaded. But on the client side grain interface assemblies may be lazily loaded later. So there is no clear point in time where we could run the codegen process for all loaded assemblies?

Is my understanding correct? If it is, then it would help to make this distinction more clear in the code.